File: first_n_primes.py
#!/usr/bin/env python3

# Python v3.8.6
#
# $ time ./first_n_primes.py 10000
# real	0m0.105s
# user	0m0.080s
# sys	0m0.013s
#
# $ time ./first_n_primes.py 100000
# real	0m2.543s
# user	0m2.462s
# sys	0m0.040s
#
# $ time ./first_n_primes.py 100000 > /dev/null
# real	0m2.492s
# user	0m2.488s
# sys	0m0.000s
#
# $ time ./first_n_primes.py 1000000 > /dev/null
# real	1m24.183s
# user	1m24.057s
# sys	0m0.037s

# PyPy v3.7.9
#
# $ time pypy3 first_n_primes.py 10000
# real	0m0.231s
# user	0m0.176s
# sys	0m0.031s
#
# $ time pypy3 first_n_primes.py 100000
# real	0m0.743s
# user	0m0.620s
# sys	0m0.063s
#
# $ time pypy3 first_n_primes.py 100000 > /dev/null
# real	0m0.675s
# user	0m0.639s
# sys	0m0.020s
#
# $ time pypy3 first_n_primes.py 1000000 > /dev/null
# real	0m15.654s
# user	0m15.563s
# sys	0m0.051s

from itertools import count
from array import array
import math
from sys import argv

isqrt = math.isqrt if hasattr(math, 'isqrt') else math.sqrt


def is_prime(n):
    # if n < 4:
    #     return n > 1
    # if n % 2 == 0 or n % 3 == 0:
    #     return False
    sqr = int(isqrt(n))

    for i in range(5, sqr + 1, 6):
        if n % i == 0 or n % (i + 2) == 0:
            return False
    return True


def first_n_primes(required_primes):
    # results = np.zeros(required_primes)
    results = array('I', [0]) * required_primes
    results[0] = 2
    results[1] = 3
    nfound = 2

    j = None
    for i in count(start=5, step=6):
        if is_prime(i):
            results[nfound] = i
            nfound += 1
            if nfound == required_primes:
                return results

        j = i + 2
        if is_prime(j):
            results[nfound] = j
            nfound += 1
            if nfound == required_primes:
                return results


if __name__ == '__main__':
    required_primes = int(argv[1])
    primes = first_n_primes(required_primes)
    primes = '\n'.join(map(str, primes))
    print(primes)

File: first-n-primes.php
#!/usr/bin/env php
<?php

/* PHP v7.14.13
 *
 * $ time ./first-n-primes.php 10000
 * real	0m0.241s
 * user	0m0.220s
 * sys	0m0.010s
 *
 * $ time ./first-n-primes.php 100000
 * real	0m7.288s
 * user	0m7.188s
 * sys	0m0.037s
 *
 * $ time ./first-n-primes.php 100000 > /dev/null
 * real	0m7.148s
 * user	0m7.125s
 * sys	0m0.010s
 */

count(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS)) or main(@$argv);

function main ($argv) {
    $desired_count = (int) @$argv[1];
    $desired_count or die();

    ob_start();
    display_first_n_primes($desired_count);
    ob_end_flush();
}

function is_prime (int $n)
{
    // if ($n < 4) { return $n > 1; }
    // if ($n % 2 === 0 || $n % 3 === 0) { return false; }

    $sqr = sqrt($n) | 0;
    for ($i = 5; $i <= $sqr; $i++) {
        if ($n % $i === 0 || $n % ($i + 2) === 0) {
            return false;
        }
    }

    return true;
}

function display_first_n_primes (int $desired_count)
{
    $desired_count-- >= 1 and print "2\n";
    $desired_count-- >= 1 and print "3\n";
    if ($desired_count <= 0) { return; }
    
    for ($i = 5, $j = 0;; $i += 6) {
        if (is_prime($i)) {
            echo $i; echo "\n";
            if (!--$desired_count) { return; }
        }

        $j = $i + 2;
        if (is_prime($j)) {
            echo $j; echo "\n";
            if (!--$desired_count) { return; }
        }
    }
}
